spec/components/cooked_post_processor_spec.rb

```diff
@@ -1146,30 +1146,41 @@
 
   context "remove direct reply full quote" do
     let(:topic) { Fabricate(:topic) }
+    let!(:post) { Fabricate(:post, topic: topic, raw: "this is the first post") }
+    let(:raw) do
+      <<~RAW
+      [quote="#{post.user.username}, post:#{post.post_number}, topic:#{topic.id}"]
+      this is the first post
+      [/quote]
+
+      and this is the third reply
+      RAW
+    end
 
     it 'works' do
-      post = Fabricate(:post, topic: topic, raw: "this is the first post")
-      hidden = Fabricate(:post, topic: topic, hidden: true, raw: "this is the second post")
-      small_action = Fabricate(:post, topic: topic, post_type: Post.types[:small_action])
-      raw = <<~RAW
-        [quote="#{post.user.username}, post:#{post.post_number}, topic:#{topic.id}"]
-        this is the first post
-        [/quote]
+      SiteSetting.remove_full_quote = true
 
-        and this is the third reply
-      RAW
+      hidden = Fabricate(:post, topic: topic, hidden: true, raw: "this is the second post after")
+      small_action = Fabricate(:post, topic: topic, post_type: Post.types[:small_action])
       reply = Fabricate(:post, topic: topic, raw: raw)
 
-      cpp = CookedPostProcessor.new(reply)
-      cpp.removed_direct_reply_full_quotes
-
+      CookedPostProcessor.new(reply).removed_direct_reply_full_quotes
       expect(topic.posts).to eq([post, hidden, small_action, reply])
       expect(reply.raw).to eq("and this is the third reply")
       expect(reply.revisions.count).to eq(1)
       expect(reply.revisions.first.modifications["raw"]).to eq([raw, reply.raw])
       expect(reply.revisions.first.modifications["edit_reason"][1]).to eq(I18n.t(:removed_direct_reply_full_quotes))
     end
 
+    it 'checks site setting' do
```

Comment:
This test deserves a better title. What about something like this?

```suggestion
    it "does nothing when 'remove_full_quote' is disabled" do
```